tiatoolbox/models/architecture/sccnn.py

```diff
@@ -0,0 +1,372 @@
+         - CoNiC
+         - 0.79
+         - 0.79
+         - 0.79
+
+    Args:
+        num_input_channels (int):
+            Number of channels in input. default=3.
+        out_height (int):
+            Output height. default=13.
+        out_width (int):
+            Output width. default=13.
+        radius (int):
+            Radius for nucleus detection, default = 12.
+        min_distance (int):
+            The minimal allowed distance separating peaks.
+            To find the maximum number of peaks, use `min_distance=1`, default=6.
+        threshold_abs (float):
+            Minimum intensity of peaks, default=0.20.
+
+    References:
+        [1] Sirinukunwattana, Korsuk, et al.
+        "Locality sensitive deep learning for detection and classification
+        of nuclei in routine colon cancer histology images."
+        IEEE transactions on medical imaging 35.5 (2016): 1196-1206.
+
+    """
+
+    def __init__(
+        self,
+        num_input_channels: int = 3,
+        out_height: int = 13,
+        out_width: int = 13,
+        radius: int = 12,
+        min_distance: int = 6,
+        threshold_abs: float = 0.20,
+    ):
+        super().__init__()
+        self.in_ch = num_input_channels
+        self.out_height = out_height
+        self.out_width = out_width
+
+        # Create mesh grid and convert to 3D vector
+        x, y = torch.meshgrid(
+            torch.arange(start=0, end=out_height),
+            torch.arange(start=0, end=out_width),
+            indexing="ij",
+        )
+
+        self.register_buffer("xv", torch.unsqueeze(x, dim=0).type(torch.float32))
+        self.register_buffer("yv", torch.unsqueeze(y, dim=0).type(torch.float32))
+
+        self.radius = radius
+        self.min_distance = min_distance
+        self.threshold_abs = threshold_abs
+
+        def conv_act_block(
+            in_channels: int, out_channels: int, kernel_size: int
+        ) -> torch.nn.ModuleDict:
+            """Convolution and Activation branch for SCCNN.
+
+            Args:
+                in_channels (int):
+                    Number of channels in input.
+                out_channels (int):
+                    Number of required channels in output.
+                kernel_size (int):
+                    Kernel size of convolution filter.
+
+            Returns:
+                torch.nn.ModuleDict:
+                    Module dictionary.
+
+            """
+            module_dict = OrderedDict()
+            module_dict["conv1"] = nn.Sequential(
+                nn.Conv2d(
+                    in_channels,
+                    out_channels,
+                    kernel_size=(kernel_size, kernel_size),
+                    stride=(1, 1),
+                    padding=0,
+                    bias=True,
+                ),
+                nn.ReLU(),
+            )
+
+            return nn.ModuleDict(module_dict)
+
+        def spatially_constrained_layer1(
+            in_channels, out_channels
```

Comment:
Type hints for these? To be consistent

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/b30c6ebf15cf23c18c850aa2ac13b35180c30786/tiatoolbox/models/architecture/sccnn.py#L155